code/game/objects/effects/decals/remains.dm

```diff
@@ -27,12 +27,26 @@
 	desc = "They look like the remains of a small reptile."
 	icon_state = "lizard"
 
-/obj/effect/decal/remains/attack_hand(mob/user as mob)
-	to_chat(user, "<span class='notice'>[src] sinks together into a pile of ash.</span>")
+//Target turns to ash or oil.
+/obj/effect/decal/remains/proc/crumble()
 	var/turf/simulated/floor/F = get_turf(src)
-	if (istype(F))
-		new /obj/effect/decal/cleanable/ash(F)
+	if (icon_state == "remainsrobot")
+		visible_message(SPAN_NOTICE("[src] degrades into a pool of oil."))
+		if (istype(F))
+			new /obj/effect/decal/cleanable/blood/oil(F)
```

Comment:
As I said a little too late on discord, I would just do a `/obj/effect/decal/remains/robot/crumble()` proc. 
Iconstate technically would work but it's not an ideal solution for this purpose. Children inherit procs from parent paths, so it's usually the better way.